Admin/BaseMediaAdmin.php

```diff
@@ -106,7 +106,7 @@ public function getNewInstance()
 
         if ($this->hasRequest()) {
             if ($this->getRequest()->isMethod('POST')) {
-                $media->setProviderName($this->getRequest()->get(sprintf('%s[providerName]', $this->getUniqid()), null, true));
+                $media->setProviderName($this->getRequest()->get(sprintf('%s', $this->getUniqid()), null, true)['providerName']);
```

Comment:
Should be in a separate commit

The commits message should explain what was wrong and how you fixed it. I don't understand what you fixed, to me if `sdlfkjsdf[providerName]` was missing, it defaults to null. Now with your change it defaults to null if `sdlfkjsdf` does not exist **but** you still use ['providerName'] on null. BTW, the [third parameter does not exist in sf 3](https://github.com/symfony/symfony/blob/3.1/src/Symfony/Component/HttpFoundation/Request.php#L717), [but exists in version 2](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/HttpFoundation/Request.php#L724) so maybe this isn't the right fix?

Maybe the right fix is : don't touch this line for sf/http-kernel < 3, and do something else for sf/http-kernel > 3